src/client/manager.ts

```diff
@@ -115,6 +116,33 @@ export class ClientManager {
         }
     }
 
+    /**
+     * Dynamically connect to a new MCP server.
+     * @param name The unique name for the new server connection.
+     * @param config The configuration for the server.
+     * @returns Promise resolving when the connection attempt is complete.
+     * @throws Error if the connection fails.
+     */
+    async connectClient(name: string, config: McpServerConfig): Promise<void> {
```

Comment:
we have the same code in https://github.com/truffle-ai/saiki/pull/57/files#diff-c2043b551fcabbf98e56573bf9459c21925e67d5a526087cf71ef5ab5636819fR93-R101 - we can update ` initializeFromConfig` to use this new function `connectClient` to avoid duplicate code